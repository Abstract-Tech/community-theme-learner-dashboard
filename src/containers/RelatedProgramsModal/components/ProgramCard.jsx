import React from 'react';
import PropTypes from 'prop-types';

import { useIntl } from '@edx/frontend-platform/i18n';
import {
  Badge,
  Card,
  // Hyperlink,
  Icon,
} from '@edx/paragon';
import { Program } from '@edx/paragon/icons';

import messages from './messages';
import './index.scss';

export const whiteFontWrapper = (node) => (<span className="text-white">{node}</span>);

export const ProgramCard = ({ data }) => {
  const { formatMessage } = useIntl();
  const numCoursesMessage = formatMessage(
    messages.courses,
    { numCourses: data.numberOfCourses },
  );
  const durationMessage = formatMessage(
    messages.duration,
    { numWeeks: data.estimatedNumberOfWeeks },
  );
  return (
    <Card
      className="program-card d-inline-block bg-primary-500 text-white pb-3.5"
      style={{ width: '18rem', color: 'white' }}
    >
      <Card.ImageCap
        className="program-card-banner"
        src={data.bannerUrl}
        srcAlt={formatMessage(messages.bannerAlt)}
        logoSrc={data.logoUrl}
        logoAlt={formatMessage(messages.logoAlt)}
      />
      <Card.Header
        title={whiteFontWrapper(data.title)}
        subtitle={whiteFontWrapper(data.provider)}
      />
      <div className="ml-4">
        <Badge variant="light" className="program-type-badge">
          <Icon src={Program} className="d-inline-block" /> {data.programType}
        </Badge>
        <div className="program-summary mt-2">
          {numCoursesMessage} â€¢ {durationMessage}
        </div>
      </div>
    </Card>
  );
};
ProgramCard.propTypes = {
  data: PropTypes.shape({
    estimatedNumberOfWeeks: PropTypes.number,
    numberOfCourses: PropTypes.number,
    bannerUrl: PropTypes.string,
    logoUrl: PropTypes.string,
    title: PropTypes.string,
    provider: PropTypes.string,
    programType: PropTypes.string,
    programUrl: PropTypes.string,
    programTypeUrl: PropTypes.string,
  }).isRequired,
};

export default ProgramCard;
