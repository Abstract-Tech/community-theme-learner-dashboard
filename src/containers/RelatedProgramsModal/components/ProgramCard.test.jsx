import React from 'react';
import { shallow } from 'enzyme';

import ProgramCard from './ProgramCard';

const props = {
  data: {
    estimatedNumberOfWeeks: 1,
    numberOfCourses: 2,
    bannerUrl: 'props.data.bannerUrl',
    logoUrl: 'props.data.logoUrl',
    title: 'props.data.title',
    provider: 'props.data.provider',
    programType: 'props.data.programType',
    programUrl: 'props.data.programUrl',
    programTypeUrl: 'props.data.programTypeUrl',
  },
};

describe('RelatedProgramsModal ProgramCard', () => {
  test('snapshot', () => {
    expect(shallow(<ProgramCard {...props} />)).toMatchSnapshot();
  });
});
