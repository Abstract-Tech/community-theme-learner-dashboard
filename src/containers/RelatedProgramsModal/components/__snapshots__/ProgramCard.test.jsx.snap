// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedProgramsModal ProgramCard snapshot 1`] = `
<Card
  className="program-card d-inline-block bg-primary-500 text-white pb-3.5"
  style={
    Object {
      "color": "white",
      "width": "18rem",
    }
  }
>
  <Card.ImageCap
    className="program-card-banner"
    logoAlt={
      <formatMessage
        msg="Provider logo"
      />
    }
    logoSrc="props.data.logoUrl"
    src="props.data.bannerUrl"
    srcAlt={
      <formatMessage
        msg="Programm banner"
      />
    }
  />
  <Card.Header
    subtitle={
      <span
        className="text-white"
      >
        props.data.provider
      </span>
    }
    title={
      <span
        className="text-white"
      >
        props.data.title
      </span>
    }
  />
  <div
    className="ml-4"
  >
    <Badge
      className="program-type-badge"
      variant="light"
    >
      <Icon
        className="d-inline-block"
      />
       
      props.data.programType
    </Badge>
    <div
      className="program-summary mt-2"
    >
      <formatMessage
        msg="{numCourses} Courses"
      />
       â€¢ 
      <formatMessage
        msg="{numWeeks} Weeks"
      />
    </div>
  </div>
</Card>
`;
